src/harness/testcases/WINNF_FT_S_REG_testcase.py

```diff
@@ -424,6 +424,196 @@ def test_WINNF_FT_S_REG_5(self):
+        'measCapability': device_10['measCapability']
+    }
+    
+    conditionals_12 = {
+        'cbsdCategory': device_12['cbsdCategory'],
+        'fccId': device_12['fccId'],
+        'cbsdSerialNumber': device_12['cbsdSerialNumber'],
+        'airInterface': device_12['airInterface'],
+        'installationParam': device_12['installationParam'],
+        'measCapability': device_12['measCapability']
+    }
+    conditionals_13 = {
+        'cbsdCategory': device_13['cbsdCategory'],
+        'fccId': device_13['fccId'],
+        'cbsdSerialNumber': device_13['cbsdSerialNumber'],
+        'airInterface': device_13['airInterface'],
+        'installationParam': device_13['installationParam'],
+        'measCapability': device_13['measCapability']
+    }
+    conditionals_14 = {
+        'cbsdCategory': device_14['cbsdCategory'],
+        'fccId': device_14['fccId'],
+        'cbsdSerialNumber': device_14['cbsdSerialNumber'],
+        'airInterface': device_14['airInterface'],
+        'installationParam': device_14['installationParam'],
+        'measCapability': device_14['measCapability']
+    }
+
+    conditionals = {
+        'registrationData': [
+            conditionals_10, conditionals_12, conditionals_13, conditionals_14
+        ]
+    }
+    self._sas_admin.PreloadRegistrationData(conditionals)
+
+    # Remove conditionals from registration
+    del device_10['cbsdCategory']
+    del device_10['airInterface']
+    del device_10['installationParam']
+    del device_10['measCapability']
+    del device_12['cbsdCategory']
+    del device_12['airInterface']
+    del device_12['measCapability']
+    del device_13['cbsdCategory']
+    del device_13['airInterface']
+    del device_13['measCapability']
+    
+    # Modify the configuration of the devices according to the specfications
+    # invalid_cbsd serial number with length > 64 octets
+    device_2['cbsdSerialNumber'] = 's' * 65
+    # invalid_fcc_id with length > 20 octets
+    device_3['fccId'] = 'f' * 21
+    device_4['userId'] = 'invalid_userId_@'
+    device_5['installationParam']['latitude'] = 90.01
+    device_6['measCapability'] = ['invalid_measCapability']
+    device_7['installationParam']['eirpCapability'] = 48
+    # Convert device_7's registration request to embed cpiSignatureData
+    convertRequestToRequestWithCpiSignature(cpi_private_key, cpi_id,
+                                            cpi_name, device_7)
+    device_8['installationParam']['latitude'] = 38.882162
+    device_8['installationParam']['longitude'] = -77.113755
+    device_8['installationParam']['height'] = 4.0
+    device_8['installationParam']['heightType'] = 'AGL'
+    device_8['installationParam']['indoorDeployment'] = False  
+    device_9['installationParam']['eirpCapability'] = 31
+    device_10['installationParam'] ={}
+    device_10['installationParam']['indoorDeployment'] = True
+    device_10['installationParam']['eirpCapability'] = 31
```

Comment:
This does not include all installationParam fields for the cpiSignedData. To ensure that we meet the criteria of the test case, we could modify the original installationParam, *not* delete them on lines 542-544, and use the same installationParam when creating CPI signed data.

Reply:
thanks, could you please read the discussion exchanged by email raised by Takashi and titled "Quick Review of PR357(REG.7)"  with test dev group.
and then we can do the modifications if needed.